pkg/plugin/kms_v2_server.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package plugin
+
+import (
+	"context"
+	"time"
+
+	"github.com/Azure/kubernetes-kms/pkg/config"
+	"github.com/Azure/kubernetes-kms/pkg/metrics"
+
+	"github.com/Azure/kubernetes-kms/pkg/version"
+	"k8s.io/klog/v2"
+	kmsv2 "k8s.io/kms/apis/v2"
+)
+
+// KeyManagementServiceV2Server is a gRPC server.
+type KeyManagementServiceV2Server struct {
+	kvClient Client
+	reporter metrics.StatsReporter
+}
```

Comment:
Why do we need another server struct?  This is the same as `KeyManagementServiceServer` but just implementing the methods for v2 server. Couldn't we just have a single server instance that implement the gRPC methods for both v1 and v2?

Reply:
The method implementation is conflicting. It fails with an error `method KeyManagementServiceServer.Encrypt already declared at`

Reply:
Go doesn't support method overloading, so I guess that makes sense!